modules/seqexec/server/src/main/scala/seqexec/server/gems/GemsControllerEpics.scala

```diff
@@ -0,0 +1,197 @@
+    val r1 = List(pause(pauseReasons), stop).flattenOption
+    // And the other way around for resuming, passing s1 (the state after stopping detectors) to startTtsg
+    val r2 = List(startCwfs(s1, cfg), resume(resumeReasons)).flattenOption
+
+    PauseResume(
+      r1.nonEmpty.option(r1.sequence.void),
+      r2.nonEmpty.option(r2.sequence.void)
+    )
+  }
+
+  override val stateGetter: GemsWfsState[F] = GemsWfsState(
+    Nested(epicsSys.apd1Active).map(DetectorStateOps.fromBoolean[Cwfs1DetectorState]).value,
+    Nested(epicsSys.apd2Active).map(DetectorStateOps.fromBoolean[Cwfs2DetectorState]).value,
+    Nested(epicsSys.apd3Active).map(DetectorStateOps.fromBoolean[Cwfs3DetectorState]).value,
+    Nested(
+      (gsaoiSys.guiding, gsaoiSys.odgw1Multiplier).mapN{case (x, y) => (x, y).mapN{case (g, m) => g && m > 0}}
+    ).map(DetectorStateOps.fromBoolean[Odgw1DetectorState]).value,
+    Nested(
+      (gsaoiSys.guiding, gsaoiSys.odgw2Multiplier).mapN{case (x, y) => (x, y).mapN{case (g, m) => g && m > 0}}
+    ).map(DetectorStateOps.fromBoolean[Odgw2DetectorState]).value,
+    Nested(
+      (gsaoiSys.guiding, gsaoiSys.odgw3Multiplier).mapN{case (x, y) => (x, y).mapN{case (g, m) => g && m > 0}}
+    ).map(DetectorStateOps.fromBoolean[Odgw3DetectorState]).value,
+    Nested(
+      (gsaoiSys.guiding, gsaoiSys.odgw4Multiplier).mapN{case (x, y) => (x, y).mapN{case (g, m) => g && m > 0}}
+    ).map(DetectorStateOps.fromBoolean[Odgw4DetectorState]).value
+  )
+
+  private def getStatusVal[A](get: F[Option[A]], name: String, system: String): F[A] = get.flatMap(
+    _.map(_.pure[F]).getOrElse((SeqexecFailure.Unexpected(s"Unable to read $name from $system.").raiseError[F, A]))
+  )
+
+  private val retrieveConfig: F[EpicsGems] = for{
+    cwfs1 <- getStatusVal(stateGetter.cwfs1, "CWFS1 active state", "GeMS RTC")
+    cwfs2 <- getStatusVal(stateGetter.cwfs2, "CWFS2 active state", "GeMS RTC")
+    cwfs3 <- getStatusVal(stateGetter.cwfs3, "CWFS3 active state", "GeMS RTC")
+    odgw1 <- getStatusVal(stateGetter.odgw1, "ODGW1 active state", "GSAOI")
+    odgw2 <- getStatusVal(stateGetter.odgw2, "ODGW2 active state", "GSAOI")
+    odgw3 <- getStatusVal(stateGetter.odgw3, "ODGW3 active state", "GSAOI")
+    odgw4 <- getStatusVal(stateGetter.odgw4, "ODGW4 active state", "GSAOI")
+  } yield EpicsGems(
+    cwfs1,
+    cwfs2,
+    cwfs3,
+    odgw1,
+    odgw2,
+    odgw3,
+    odgw4
+  )
+
+  private def pause(pauseConditions: PauseConditionSet)
+  : Option[F[Unit]] = {
+    val unguided = pauseConditions.contains(PauseCondition.GaosGuideOff).option(UnguidedCondition)
+    val offset = pauseConditions.offsetO.as(OffsetCondition)
+    val instMove = pauseConditions.contains(PauseCondition.InstConfigMove).option(InstrumentCondition)
+
+    val reasons = List(unguided, offset, instMove).flattenOption
+
+    reasons.nonEmpty.option {
+      epicsSys.LoopControl.setCommand(PauseCmd) *>
+        epicsSys.LoopControl.setReasons(reasons.mkString("|")) *>
```

Comment:
Is this format required on the Epics side?

Reply:
Yes